src/d3a/models/state.py

```diff
@@ -56,17 +56,33 @@ def get_state(self) -> Dict:
 
     @abstractmethod
     def restore_state(self, state_dict: Dict):
-        pass
+        """Update the state of the device using the provided dictionary."""
 
     @abstractmethod
     def delete_past_state_values(self, time_slot):
```

Comment:
While you are on it, can you please add a type to `time_slot` here (i think it is pendulum.DateTime) ?
I find this very helpful. Thanks a lot!